src/DonationForms/DonationFormsAdminPage.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Give\DonationForms;
+
+/**
+ * @unreleased
+ */
+class DonationFormsAdminPage
+{
+    /**
+     * @unreleased
+     */
+    public function addFormSubmenuLink()
+    {
+        global $submenu;
+
+        $giveParentPageKey = 'edit.php?post_type=give_forms';
+
+        if ( ! isset($submenu[$giveParentPageKey])) {
+            return;
+        }
+
+        /** @var $giveSubmenu array [0: "Add Form", 1: "edit_give_forms", 2: "post-new.php?post_type=give_forms", ...] */
+        foreach ($submenu[$giveParentPageKey] as $giveSubmenuKey => $giveSubmenu) {
+            if ($this->isLegacyAddFormSubmenuLink($giveSubmenu)) {
+                $submenu[$giveParentPageKey][$giveSubmenuKey][2] = 'edit.php?post_type=give_forms&page=givewp-form-builder';
+            }
+        }
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param $giveSubmenu array [0: "Add Form", 1: "edit_give_forms", 2: "post-new.php?post_type=give_forms", ...]
+     */
+    private function isLegacyAddFormSubmenuLink(array $giveSubmenu): bool
+    {
+        return isset($giveSubmenu[2]) && 'post-new.php?post_type=give_forms' === $giveSubmenu[2];
+    }
```

Comment:
WordPress provides tools to handle this beautifully without getting your hands dirty by changing the global variable directly. All you’ll need are these 2 lines of code:

```php
remove_submenu_page('edit.php?post_type=give_forms', 'post-new.php?post_type=give_forms');
add_submenu_page('edit.php?post_type=give_forms', 'Add Form', 'Add Form', 'edit_posts', 'edit.php?post_type=give_forms&page=givewp-form-builder');
```

Reply:
Oh man, much better! Thanks for the suggestion. Resolved in 04265c3aeb44e0bb9a45ebab8aee2b688470efd6